docs/components/date_pickers.md

```diff
@@ -0,0 +1,320 @@
+
+Common use cases include:
+
+- Making a restaurant reservation
+- Scheduling a meeting
+
+They aren’t ideal for selecting dates in the distant past or future that require more navigation, such as entering a birth date or expiration date.
+
+Mobile date pickers allow selection of a single date and year.
+
+### Date pickers example
+
+`showDatePicker`
+- [Class definition](https://api.flutter.dev/flutter/material/showDatePicker.html)
+- [GitHub souce](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/pickers/date_picker_dialog.dart)
+- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/date_pickers/regular&gh_ref=develop)
+
+The following example shows a date picker.
+
+![Date picker with date set to November 17, 2020](assets/date_pickers/date_picker_example.png)
+
+```dart
+final DateTime newDate = await showDatePicker(
+  context: context,
+  initialDate: DateTime(2020, 11, 17),
+  firstDate: DateTime(2017, 1),
+  lastDate: DateTime(2022, 7),
+  helpText: 'Select a date',
+);
+```
+
+### Anatomy and key properties
+
+The following diagram shows the elements of a date picker:
+
+![Date pickers anatomy diagram](assets/date_pickers/pickers_anatomy.png)
+
+1. Title
+1. Selected date
+1. Switch-to-keyboard input icon
+1. Year selection menu
+1. Month pagination
+1. Current date
+1. Selected date
+
+#### Title
+
+&nbsp;                                  | Attribute                
+--------------------------------------- | ------------------------
+**Text** | `helpText` property in `showDatePicker`
+
+#### Selected date
+
+&nbsp;                                  | Attribute                
+--------------------------------------- | ------------------------
+**Date** | `initialDate` property in `showDatePicker`
+
+#### Current date
+
+&nbsp;                                  | Attribute                
+--------------------------------------- | ------------------------
+**Date** | `currentDate` property in `showDatePicker`
+
+## Date range pickers
+
+Mobile date range pickers allow selection of a range of dates. They cover the entire screen.
+
+Common use cases include:
+
+- Booking a flight
+- Reserving a hotel
+
+### Date range picker example
+
+`showDateRangePicker`
+- [Class definition](https://api.flutter.dev/flutter/material/showDateRangePicker.html)
+- [GitHub souce](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/pickers/date_range_picker_dialog.dart)
+- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/date_pickers/range&gh_ref=develop)
+
+The following example shows a date range picker.
+
+![Date range picker with dates set from November 17, 2020 to November 24, 2020](assets/date_pickers/date_range_picker_example.png)
```

Comment:
Please update alt-text to:

"Date range picker selecting November 17, 2020 to November 24, 2020, current date of November 13, 2020."